src/chain/chain.rs

```diff
@@ -628,6 +628,49 @@ impl Chain {
             .map(|member_info| &member_info.connection_info)
     }
 
+    /// Returns the `ConnectionInfo` for a `XorName` in our section.
+    // WIP: this is probably unnecessary slow
+    #[cfg(feature = "mock_base")]
+    pub fn get_member_name_connection_info(&self, name: &XorName) -> Option<ConnectionInfo> {
+        self.state
```

Comment:
Agree, I'll update the map to take a `XorName` as key and `MembersInfo` to also contain a `P2pNode` instead of a `ConnectionInfo`. That way we can to straight lookups and return references